### ISSUE TITLE
Refactor problematic markup for 'upload image' feature in post editor

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#13947]
In the post editor, we allow a user to upload:

- a cover image via the "Add a cover image" button
- any other images via the "Upload image" button

In both cases, the HTML markup is a `button` element with an `input` descendant. It's not valid HTML to nest interactive items inside each other like this, and this [can cause accessibility issues with screen reading software](https://dequeuniversity.com/rules/axe/4.2/nested-interactive?application=AxeChrome).

The nesting in this particular case is also causing some unfortunate accessible names on these button elements, for example:

![screenshot showing the 'upload image' button is announced via VoiceOver as "upload image upload image upload an image button"](https://user-images.githubusercontent.com/20773163/121344633-e5b3d500-c91b-11eb-91a9-23af12b07f46.png)

## Expected behavior after refactor

- The buttons no longer have inputs as descendants
- The current visual style is maintained
- Accessible names for the controls are logical and announced via screen reader
- Mobile functionality, including that in the mobile app, are unaffected (NB: the app handles image upload natively, and we need to be careful not to regress this behavior. Any PR addressing this will need tested in both desktop/mobile web and in the native app)
- "Upload image" functionality works by both keyboard and mouse

## Other context

After an upgrade to `axe-core`, this markup started causing our frontend tests to fail in the following files:
- `app/javascript/article-form/components/__tests__/ImageUploader.test.jsx`
- `app/javascript/article-form/components/__tests__/ArticleCoverImage.test.jsx` 
- `app/javascript/article-form/components/__tests__/Form.test.jsx`

The fix for this issue should reinstate the normal axe rules for this test.

**Comment by mstruve** [comment]
@aitchiss can you put a size tag on this based on what size you think it would be? Please!
